src/init.cpp

```diff
@@ -1918,17 +1919,25 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
     connOptions.vSeedNodes = args.GetArgs("-seednode");
 
-    // Initiate outbound connections unless connect=0
-    connOptions.m_use_addrman_outgoing = !args.IsArgSet("-connect");
-    if (!connOptions.m_use_addrman_outgoing) {
-        const auto connect = args.GetArgs("-connect");
+    const auto connect = args.GetArgs("-connect");
+    if (!connect.empty() || args.IsArgNegated("-connect")) {
+        // Do not initiate other outgoing connections when connecting to trusted
+        // nodes, or when -noconnect is specified.
+        connOptions.m_use_addrman_outgoing = false;
+
+        // Assign addresses of trusted nodes, if any.
         if (connect.size() != 1 || connect[0] != "0") {
             connOptions.m_specified_outgoing = connect;
         }
+
+        // Warn that -seednode setting will be ignored if trusted nodes were
+        // specified.
         if (!connOptions.m_specified_outgoing.empty() && !connOptions.vSeedNodes.empty()) {
             LogPrintf("-seednode is ignored when -connect is used\n");
         }
 
+        // Warn that -dnsseed will be ignored if automatic connections are
+        // disabled.
         if (args.IsArgSet("-dnsseed") && args.GetBoolArg("-dnsseed", DEFAULT_DNSSEED) && args.IsArgSet("-proxy")) {
             LogPrintf("-dnsseed is ignored when -connect is used and -proxy is specified\n");
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/30529#discussion_r1815689215

> Better to explain _why_ than _what_ (log message already explains _what_). To me it's not immediately obvious that we should avoid using DNS seeds when having a Tor SOCKS5 proxy?

Just dropped the new comments for now. There should be more chances to make improvements here later.

The comment was saying why the dnsseed is ignored: because automatic connections are disabled. The log message says the seed is ignored because -connect is specified, but it is not obvious (IMO) that specifying -connect disables other connections. 

The -proxy part of this check seems buggy to me and I mentioned that in the commit message, but didn't look into the history, and in any case don't want to change the behavior in this PR.

Reply:
Unintended negation in commit message of 0a954ce80cf9d9744eaf213e3b81d4016b19bf88?
"Not sure if it really makes sense >not< to warn about seednode being ignored if
-noconnect is specified, and only to warn about -dnsseed being ignored when
-proxy is specified, but these behaviors are not changed from before."